services/vso/vso.tester.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+const Joi = require('joi')
+const ServiceTester = require('../service-tester')
+const { isBuildStatus } = require('../test-validators')
+
+const t = new ServiceTester({
+  id: 'vso',
+  title: 'Visual Studio Team Services',
+})
+module.exports = t
+
+// Builds
+
+t.create('build status on default branch')
+  .get('/build/devdiv/0bdbc590-a062-4c3f-b0f6-9383f67865ee/7716.json')
+  .expectJSONTypes(
+    Joi.object().keys({
+      name: 'build',
+      value: isBuildStatus,
+    })
+  )
+
+t.create('build status on named branch')
+  .get('/build/devdiv/0bdbc590-a062-4c3f-b0f6-9383f67865ee/7716/master.json')
+  .expectJSONTypes(
+    Joi.object().keys({
+      name: 'build',
+      value: isBuildStatus,
+    })
+  )
+
+t.create('build status on unknown repo')
+  .get('/build/this-repo/does-not/exist.json')
+  .expectJSON({ name: 'build', value: 'inaccessible' })
+
+t.create('build status with connection error')
+  .get('/build/foo/bar/foobar.json')
+  .networkOff()
+  .expectJSON({ name: 'build', value: 'inaccessible' })
+
+// Releases
+
+t.create('release status is succeeded')
+  .get('/release/devdiv/0bdbc590-a062-4c3f-b0f6-9383f67865ee/77/16.json')
+  .intercept(nock =>
+    nock('https://devdiv.vsrm.visualstudio.com')
```

Comment:
Out of interest, why do we have to mock the responses for the release badges?

Reply:
The reason is quite simple: I didn't find any public VSTS release like the one I found for builds (from Microsoft .NET Core). Since I prefer not to use one from my company (since it is private, hence it can change without notice).